### DISCUSSION TITLE
List orphaned fields when deleting a Craft 4.x Matrix field

### DISCUSSION THREAD

**Discussion by proimage** [craftcms/cms#9690] [Ideas] [answered]
This is basically the equivalent of https://github.com/spicywebau/craft-neo/issues/490, but for the anticipated revamped Matrix field that's supposed to come with Craft 4.0 ("M4trix"?), instead of for Neo.

Basically, when deleting a M4trix field, it would be very useful to be shown a list of all the other custom fields that were only used in that M4trix field, and nowhere else.

**Comment by brandonkelly** [comment, marked as answer]
It’s a little early to be thinking about this… Matrix fields contain each of their sub-fields’ definitions, so they all get deleted alongside the parent Matrix field.

If/when we make it possible to select globally-defined fields within Matrix fields (like Neo), this would make sense.

**Reply by proimage** [reply]
> If/when we make it possible to select globally-defined fields within Matrix fields (like Neo), this would make sense.

Yeah, that's what I meant by Craft 4.x's Matrix. Sorry if it wasn't clear.

**Comment by alexanderbuergin** [comment]
Use Neo as a core Craft CMS field instead of the Matrix field approach.
